les/peer.go

```diff
@@ -876,18 +879,25 @@ func (p *clientPeer) sendAnnounce(request announceData) error {
 	return p2p.Send(p.rw, AnnounceMsg, request)
 }
 
+// allowInactive implements clientPoolPeer
+func (p *clientPeer) allowInactive() bool {
+	return false
+}
+
 // updateCapacity updates the request serving capacity assigned to a given client
 // and also sends an announcement about the updated flow control parameters
 func (p *clientPeer) updateCapacity(cap uint64) {
 	p.lock.Lock()
 	defer p.lock.Unlock()
 
-	p.fcParams = flowcontrol.ServerParams{MinRecharge: cap, BufLimit: cap * bufLimitRatio}
-	p.fcClient.UpdateParams(p.fcParams)
-	var kvList keyValueList
-	kvList = kvList.add("flowControl/MRR", cap)
-	kvList = kvList.add("flowControl/BL", cap*bufLimitRatio)
-	p.queueSend(func() { p.sendAnnounce(announceData{Update: kvList}) })
+	if cap != p.fcParams.MinRecharge {
```

Comment:
If the capacity is raised, then after a while it's reset back to the default capacity(which is equal with minimal capacity). In this case we can't nofity the change

Reply:
I am not sure if I understand correctly, maybe it's a misunderstanding. `p.fcParams.MinRecharge` is not `minCapacity`, it is the "minimal buffer recharge rate" which is basically the current capacity. So if the capacity has previously been raised then it is at a higher value and therefore the revert to `minCapacity` should also work properly.

Reply:
I see! `p.fcParams.MinRecharge` is basically the current `recharge rate` (kind of current capacity). I was misunderstood previous.